### PR TITLE
RD-4478 stronger apigw source filter

### PR DESCRIPTION
**Pull request by OrrLumigo**

### CODE REVIEW

File: src/test/unit/test_tracer.py [hunk 1]
```diff
@@ -51,9 +53,25 @@ def lambda_test_function(event, context):
...
+    """
+
+    @lumigo_tracer(token=token)
+    def lambda_test_function(event, context):
+        pass
+
+    lambda_test_function({}, context)
+    captured = capsys.readouterr()
+    expected = "Lumigo Warning: Invalid token used, copy your token from Settings → Tracing from Lumigo’s platform\n"
+    assert captured[0] == expected
```
Comment (saartochner) [inline src/test/unit/test_tracer.py:69]: please also check that the tracer haven't send any span to Lumigo

File: src/lumigo_tracer/lumigo_utils.py [hunk 2]
```diff
@@ -156,12 +156,17 @@ def config(
     :param edge_kinesis_aws_access_key_id: The credentials to push to the Kinesis in China region
     :param edge_kinesis_aws_secret_access_key: The credentials to push to the Kinesis in China region
     """
+    Configuration.token = token or os.environ.get(LUMIGO_TOKEN_KEY, "")
+    if not (Configuration.token and re.match("[t][_][[a-z0-9]{21}", Configuration.token)):
+        warn_client(
+            "Invalid token used, copy your token from Settings → Tracing from Lumigo’s platform"
```
Comment (saartochner) [inline src/lumigo_tracer/lumigo_utils.py:162]: The `->` character will not work in cloudwatch.
Please change it to `>`

Comment (saartochner) [reply src/lumigo_tracer/lumigo_utils.py:162]: Either way, change to 
`Lumigo Warning: Invalid Token. Go to Lumigo Settings to get a valid token. `
So we could change the structure in the future :)

**Review by saartochner — CHANGES_REQUESTED** [review]
Well done!

File: src/test/unit/test_tracer.py (hunk 1 above)
Comment (OrrLumigo) [reply src/test/unit/test_tracer.py:69]: what is span?

Comment (saartochner) [reply src/test/unit/test_tracer.py:69]: `reporter_mock.call_count == 2`
And use the fixture `reporter_mock`

**Review by saartochner — APPROVED** [review]
Awesome!

**Pull request merged**